src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContextProvider.cs

```diff
@@ -64,7 +67,7 @@ public void ReleaseProjectContext(AggregateCrossTargetProjectContext context)
         {
             Requires.NotNull(context, nameof(context));
 
-            lock (_gate)
+            using (_gate.DisposableWait())
```

Comment:
Looks like you can use the async form here and flow the task backwards.

Reply:
I've added async versions for these. As you said, it didn't end up spreading the virus very far :)